### PR TITLE
Issue #505 Dry out #get and #put methods as they have identicall parts

### PR DESCRIPTION
**Pull request by dapi**
#505

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#570](https://github.com/zold-io/zold/issues/570) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#570](https://github.com/zold-io/zold/issues/570) is assigned to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa), here is [why](https://www.0crat.com/footprint/CAZPZR9FS/845aa830-5934-47f8-89ed-3820c477e7e2); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by emilianodellacasa** [comment]
@dapi I cannot see any issue on this PR @yegor256 we can merge

File: lib/zold/http.rb [hunk 1]
```diff
@@ -129,6 +116,26 @@ def headers
...
+    end
+
+    def base_url
+      "#{@uri.scheme}://#{@uri.host}:#{@uri.port}"
+    end
+
+    def path
+      path = @uri.path
+      path += '?' + @uri.query if @uri.query
+
```
Comment (yegor256) [inline lib/zold/http.rb]: @dapi we don't like this, check: https://www.yegor256.com/2014/11/03/empty-line-code-smell.html

**Comment by yegor256** [comment]
@dapi one small comment from me, above

File: lib/zold/http.rb (hunk 1 above)
Comment (dapi) [reply lib/zold/http.rb]: Ok. You need to describe it with rubocop rules.

Comment (dapi) [reply lib/zold/http.rb]: What do you think about this solution?

`[@uri.path, @ury.query].compact.join(‘?’)`

**Comment by dapi** [comment]
Soon be other solution

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#570](https://github.com/zold-io/zold/issues/570) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256)

**Comment by ypshenychka** [comment]
@emilianodellacasa  According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

No issue were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by ypshenychka** [comment]
@dapi According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>Pull request description explains the solution proposed and contains a link to the original ticket it is related to

Please extend description with more details.

**Comment by ypshenychka** [comment]
@dapi According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>Messages in a job [always](https://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html#4-avoid-noiseaddress-your-comments) start with a name of a user they are addressed to

Please correct your [message](https://github.com/zold-io/zold/pull/570#issuecomment-441931719) and code review comments ([here](https://github.com/zold-io/zold/pull/570#discussion_r236397006) and [here](https://github.com/zold-io/zold/pull/570#discussion_r236397923)) by indicating an addressee in the beginning.

**Comment by ypshenychka** [comment]
@emilianodellacasa thanks

**Comment by ypshenychka** [comment]
@dapi ping

**Comment by dapi** [comment]
@ypshenychka comments are not actual. code is deeply changed.

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request closed without merging**